builtin/mainmenu/dlg_config_world.lua

```diff
@@ -42,15 +40,46 @@ local function get_formspec(data)
 		retval = retval ..
 			"textarea[0.25,0.7;5.75,7.2;;" .. info .. ";]"
 	else
+		local hard_deps, soft_deps = pkgmgr.get_dependencies(mod.path)
+		local hard_deps_str = table.concat(hard_deps, ",")
+		local soft_deps_str = table.concat(soft_deps, ",")
+
 		retval = retval ..
 			"label[0,0.7;" .. fgettext("Mod:") .. "]" ..
-			"label[0.75,0.7;" .. mod.name .. "]" ..
-			"label[0,1.25;" .. fgettext("Dependencies:") .. "]" ..
-			"textlist[0,1.75;5,2.125;world_config_depends;" .. hard_deps ..
-			";0]" ..
-			"label[0,3.875;" .. fgettext("Optional dependencies:") .. "]" ..
-			"textlist[0,4.375;5,1.8;world_config_optdepends;" ..
-			soft_deps .. ";0]"
+			"label[0.75,0.7;" .. mod.name .. "]"
+
+		if hard_deps_str == "" then
+			if soft_deps_str == "" then
+				retval = retval ..
+					"label[0,1.25;" ..
+					fgettext("No (optional) dependencies") .. "]"
+			else
+				retval = retval ..
+					"label[0,1.25;" .. fgettext("No hard dependencies") ..
+					"]" ..
+					"label[0,1.75;" .. fgettext("Optional dependencies:") ..
+					"]" ..
+					"textlist[0,2.1;5,4;world_config_optdepends;" ..
```

Comment:
Label and textlist need more spacing, otherwise it looks OK now.
![grafik](https://user-images.githubusercontent.com/1497498/45174727-def85080-b20b-11e8-8568-2a30b2005d72.png)